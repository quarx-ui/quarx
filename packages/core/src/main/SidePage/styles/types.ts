import { ModalStyleParams } from '@core';

export type SidePageStyleParams = Omit<ModalStyleParams, 'scrollBehaviour'>;
