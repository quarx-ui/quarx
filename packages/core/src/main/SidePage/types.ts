import { ComponentPropsWithHTML, ModalPropsWithoutHTML, ModalStyleParams, WithClassesAndStyles } from '@core';
import { SidePageStyleKeys } from '@core/src/main/SidePage/styles';
import { ModalCSSVarKeys } from '@core/src/main/Modal/styles';

export type SidePagePropsWithoutHtml =
    & Omit<ModalPropsWithoutHTML, 'scrollBehaviour' | 'styles'>
    & WithClassesAndStyles<SidePageStyleKeys, ModalStyleParams, ModalCSSVarKeys>

export type SidePageProps = ComponentPropsWithHTML<SidePagePropsWithoutHtml>;
