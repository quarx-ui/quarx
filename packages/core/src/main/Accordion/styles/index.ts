import { KeysFromUseStyles, makeStyles, sizeToHeadlineSize, typography } from '@core/styles';
import { QX_SIZE } from '@core/enums';
import { AccordionStyleParams } from './types';
import { sizeToAccordionPadding, sizeToTextStyle } from './map';

export const useStyles = makeStyles((
    { palette, transitions },
    {
        open,
        size,
        showDivider,
        inheritTextStyles,
    }: AccordionStyleParams,
) => ({
    root: [
        { padding: `${sizeToAccordionPadding[size]}px 0` },
        showDivider && { paddingBottom: 0 },
    ],
    summary: {
        cursor: 'pointer',
    },
    headline: {
        display: 'flex',
        alignItems: 'center',
        justifyContent: 'space-between',
    },
    title: {
        color: palette.text.main,
        ...typography.Headline[sizeToHeadlineSize[size]].Semibold,
    },
    icons: {
        display: 'inline-grid',
        gridAutoFlow: 'column',
        gap: '14px',
    },
    collapseIcon: [
        {
            display: 'flex',
            alignItems: 'center',
            justifyContent: 'center',
            transition: transitions.create('transform'),
        },
        open && { transform: 'rotate(-180deg)' },
    ],
    statusIcon: {
        display: 'flex',
        alignItems: 'center',
        justifyContent: 'center',
    },
    description: {
        color: palette.text.secondary,
        ...sizeToTextStyle[size],
    },
    details: [
        {
            marginTop: size === QX_SIZE.xSmall ? 12 : 16,
        },
        !inheritTextStyles && sizeToTextStyle[size],
    ],
    divider: {
        marginTop: sizeToAccordionPadding[size] - 1,
    },
}));

export type AccordionStyleKeys = KeysFromUseStyles<typeof useStyles>;
