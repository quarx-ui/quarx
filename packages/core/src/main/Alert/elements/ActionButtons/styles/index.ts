import { KeysFromUseStyles, makeStyles } from '@core';
import { ALERT_TYPE } from '@core/src/main/Alert';
import { ActionButtonsStyleParams } from './types';

export const useStyles = makeStyles((
    _,
    { type }: ActionButtonsStyleParams,
) => ({
    root: [
        type === 'horizontal' && {
            marginLeft: 10,
        },
    ],
    button: [
        type === ALERT_TYPE.horizontal && {
            paddingTop: 7,
            paddingBottom: 7,
        },
    ],
}));

export type ActionButtonStyleKeys = KeysFromUseStyles<typeof useStyles>;
