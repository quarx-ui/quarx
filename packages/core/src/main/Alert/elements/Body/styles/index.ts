import { KeysFromUseStyles, makeStyles, paramsToCss, typography } from '@core';
import { BodyStyleParams } from './types';

export const useStyles = makeStyles((
    { palette },
    { size }: BodyStyleParams,
) => ({
    root: {
        width: '100%',
    },
    texts: {},
    title: paramsToCss(size)({
        small: typography.Text.M.Semibold,
        large: typography.Text.L.Semibold,
    }),
    description: [
        {
            textAlign: 'justify',
            margin: 0,
            color: palette.text.secondary,
        },
        paramsToCss(size)({
            small: typography.Text.M.Regular,
            large: typography.Text.L.Regular,
        }),
    ],

    actions: paramsToCss(size)({
        small: { marginTop: 16 },
        large: { marginTop: 20 },
    }),
}));

export type BodyStyleKeys = KeysFromUseStyles<typeof useStyles>;
