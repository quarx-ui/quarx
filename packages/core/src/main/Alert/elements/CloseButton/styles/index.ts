import { KeysFromUseStyles, makeStyles } from '@core';
import { ALERT_TYPE } from '@core/src/main/Alert';
import { CloseButtonStyleParams } from './types';

export const useStyles = makeStyles((
    { palette },
    { type }: CloseButtonStyleParams,
) => ({
    root: {
        height: 40,
        width: 40,
        minWidth: 40,
        display: 'flex',
        justifyContent: 'center',
        alignItems: 'center',
        marginLeft: 12,
    },
    button: [
        type === ALERT_TYPE.horizontal && {
            color: palette.colors.secondary.border,
        },
    ],
}));

export type CloseButtonStyleKeys = KeysFromUseStyles<typeof useStyles>;
