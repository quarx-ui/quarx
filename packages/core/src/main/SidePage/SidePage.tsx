import { css } from '@emotion/react';
import { FC, forwardRef } from 'react';
import { Styles, usePropsOverwrites } from '@core/styles';
import { OVER_SCREEN_APPEARANCE, OVER_SCREEN_PLACEMENT, Modal } from '@core/src';
import { MODAL_SCROLL_BEHAVIOR } from '@core/src/main/Modal/constants';
import { FooterBlockStyleKeys } from '@core/src/system/FooterBlock/styles';
import clsx from 'clsx';
import { SidePageProps } from './types';
import { useStyles } from './styles';

export const SidePage: FC<SidePageProps> = forwardRef<HTMLDivElement, SidePageProps>((
    initialProps,
    ref,
) => {
    const { props, styleProps, cn } = usePropsOverwrites('SidePage', initialProps);

    const styles = useStyles(styleProps);

    const footerBlockExternalStyles = props.FooterProps?.styles as Styles<FooterBlockStyleKeys>;

    return (
        <Modal
            {...props}
            ref={ref}
            scrollBehaviour={MODAL_SCROLL_BEHAVIOR.body}
            OverScreenProps={{
                appearance: OVER_SCREEN_APPEARANCE.slide,
                placement: OVER_SCREEN_PLACEMENT.right,
                ...props.OverScreenProps,
            }}
            css={styles.root}
            className={cn('root')}
            styles={{
                ...styleProps.styles,
                box: styles.box,
                scrollContainer: styles.scrollContainer,
            }}
            classes={{
                ...props.classes,
                box: clsx(cn('box'), props.classes?.box),
                scrollContainer: clsx(cn('scrollContainer'), props.classes?.scrollContainer),
            }}
            FooterProps={{
                ...props.FooterProps,
                styles: {
                    ...props.FooterProps?.styles,
                    root: css([styles.footerBlock, footerBlockExternalStyles?.root]),
                },
                classes: {
                    ...props.FooterProps?.classes,
                    root: clsx(cn('footerBlock'), props.FooterProps?.classes?.root),
                },
            }}
        />
    );
});
