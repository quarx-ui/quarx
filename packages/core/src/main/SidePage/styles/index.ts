import {
    KeysFromUseStyles,
    makeStyles,
} from '@core/styles';
import { ModalStyleKeys } from '@core/src/main/Modal/styles';

export const useStyles = makeStyles(() => ({
    root: {},
    scrollContainer: {
        flexGrow: 'unset',
    },
    box: [
        {
            borderRadius: 'unset',
            margin: 0,
            marginLeft: 'auto',
            height: '100%',
            width: 400,
            maxHeight: 'unset',
        },
    ],
    footerBlock: {
        marginTop: 'auto',
    },
}));

export type SidePageStyleKeys = KeysFromUseStyles<typeof useStyles> | ModalStyleKeys;
export * from './types';
