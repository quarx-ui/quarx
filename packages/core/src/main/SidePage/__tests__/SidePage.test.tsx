import React, { FC, useState } from 'react';
import { render, screen } from '@testing-library/react';
import { testStyleParams } from '@core/test-utils';
import { Button, SidePageProps, SidePage } from '@core';
import userEvent from '@testing-library/user-event';
import { SidePageStyleParams } from '@core/src/main/SidePage/styles';

const buttonTestId = 'buttonTestId';

const Template: FC<SidePageProps> = (props) => {
    const [isOpened, setIsOpened] = useState(false);

    return (
        <div>
            <Button
                onClick={() => setIsOpened(true)}
                data-testid={buttonTestId}
            />
            <SidePage
                open={isOpened}
                {...props}
            />
        </div>
    );
};

describe('SidePage', () => {
    testStyleParams<Partial<SidePageStyleParams>, SidePageProps>(
        SidePage,
        {
            size: 'medium',
        },
        { open: true },
        { capture: 'body', qxClassname: 'QxModal' },
    )({
        size: ['small', 'medium'],
    });

    it('sidePage in the document', async () => {
        const { asFragment } = render(
            <Template
                title="Title"
                subTitle="SubTitle"
                body="Body"
                footerButtons={{
                    success: {
                        children: 'Success',
                    },
                    secondary: {
                        children: 'Secondary',
                    },
                    danger: {
                        children: 'Danger',
                    },
                }}
            />,
        );

        const button = screen.getByTestId(buttonTestId);

        expect(asFragment().ownerDocument.body).toMatchSnapshot();
        userEvent.click(button);
        expect(asFragment().ownerDocument.body).toMatchSnapshot();
    });

    it('keepMounted sidePage always in the document', async () => {
        const { asFragment } = render(<Template
            title="Title"
            subTitle="SubTitle"
            body="Body"
            OverScreenProps={{
                keepMounted: true,
            }}
        />);

        const button = screen.getByTestId(buttonTestId);

        expect(asFragment().ownerDocument.body).toMatchSnapshot();
        userEvent.click(button);
        expect(asFragment().ownerDocument.body).toMatchSnapshot();
    });
});
