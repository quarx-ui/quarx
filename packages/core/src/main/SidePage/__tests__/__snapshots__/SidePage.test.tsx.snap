// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SidePage keepMounted sidePage always in the document 1`] = `
<body
  style=""
>
  <div>
    <div>
      <button
        class="QxBaseButton QxBaseButton_size_medium QxBaseButton_color_brand QxBaseButton_type_contained QxBaseButton_borderRadius_medium QxButton QxButton_size_medium css-3dyhpy-QxBaseButton-root-QxButton-root"
        data-testid="buttonTestId"
        type="button"
      />
    </div>
  </div>
  <div
    aria-hidden="true"
    class="QxOverScreen QxOverScreen_placement_right QxModal QxModal_size_medium QxModal_scrollBehaviour_body QxModal_hasHeader QxSidePage css-1t42n66-QxOverScreen-root-QxModal-root-QxSidePage-root"
    data-component="QxOverScreen"
    data-index="0"
    datatype="overscreen"
    role="presentation"
    tabindex="0"
  >
    <div
      aria-hidden="true"
      class="QxBackdrop css-s2jrcg-QxBackdrop-root-QxBackdrop-root"
      data-component="QxBackdrop"
    />
    <div
      class="QxModal-scrollContainer QxSidePage-scrollContainer css-1i0aqxr-QxSidePage-scrollContainer-QxModal-scrollContainer-QxSidePage-scrollContainer-QxModal-scrollContainer"
      data-component="QxOverScreen-child"
    >
      <div
        class="QxModal-box QxSidePage-box css-3oh5up-QxSidePage-box-QxModal-box"
      >
        <div
          class="QxHeaderBlock QxHeaderBlock_size_medium css-9oje8r-QxHeaderBlock-root"
        >
          <div
            class="QxHeaderBlock-content css-8aatdw-QxHeaderBlock-content"
          >
            <div
              class="QxHeaderBlock-title css-1p40syo-QxHeaderBlock-title"
            >
              Title
            </div>
            <div
              class="QxHeaderBlock-subTitle css-12h5hex-QxHeaderBlock-subTitle"
            >
              SubTitle
            </div>
          </div>
          <button
            class="QxBaseButton QxBaseButton_size_small QxBaseButton_color_secondary QxBaseButton_type_text QxBaseButton_borderRadius_medium QxIconButton QxIconButton_size_small QxHeaderBlock-closeButton css-yj9ie1-QxBaseButton-root-QxIconButton-root-QxHeaderBlock-closeButton"
            type="button"
          >
            <div
              class="QxIconButton-icon css-10kmd5c-QxIconButton-icon"
            >
              <svg
                fill="none"
                height="24"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M7.70714 17.7071C7.31661 18.0976 6.68345 18.0976 6.29292 17.7071C5.9024 17.3166 5.9024 16.6834 6.29292 16.2929L10.5858 12L6.29293 7.70711C5.9024 7.31658 5.9024 6.68342 6.29293 6.29289C6.68345 5.90237 7.31662 5.90237 7.70714 6.29289L12 10.5858L16.2929 6.29289C16.6834 5.90237 17.3166 5.90237 17.7071 6.29289C18.0977 6.68342 18.0977 7.31658 17.7071 7.70711L13.4142 12L17.7071 16.2929C18.0977 16.6834 18.0977 17.3166 17.7071 17.7071C17.3166 18.0976 16.6834 18.0976 16.2929 17.7071L12 13.4142L7.70714 17.7071Z"
                  fill="currentColor"
                />
              </svg>
            </div>
          </button>
        </div>
        <div
          class="QxModal-body css-r81a94-QxModal-body"
        >
          Body
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`SidePage keepMounted sidePage always in the document 2`] = `
<body
  style="padding-right: 1024px; overflow: hidden;"
>
  <div>
    <div>
      <button
        class="QxBaseButton QxBaseButton_size_medium QxBaseButton_color_brand QxBaseButton_type_contained QxBaseButton_borderRadius_medium QxButton QxButton_size_medium css-3dyhpy-QxBaseButton-root-QxButton-root"
        data-testid="buttonTestId"
        type="button"
      />
    </div>
  </div>
  <div
    aria-hidden="false"
    class="QxOverScreen QxOverScreen_placement_right QxOverScreen_mounted QxModal QxModal_size_medium QxModal_scrollBehaviour_body QxModal_hasHeader QxSidePage css-1tn48qf-QxOverScreen-root-QxModal-root-QxSidePage-root"
    data-component="QxOverScreen"
    data-index="0"
    datatype="overscreen"
    role="presentation"
    tabindex="0"
  >
    <div
      aria-hidden="false"
      class="QxBackdrop css-lak5sv-QxBackdrop-root-QxBackdrop-root enter-active"
      data-component="QxBackdrop"
    />
    <div
      class="QxModal-scrollContainer QxSidePage-scrollContainer css-zezqa0-QxSidePage-scrollContainer-QxModal-scrollContainer-QxSidePage-scrollContainer-QxModal-scrollContainer enter-active"
      data-component="QxOverScreen-child"
    >
      <div
        class="QxModal-box QxSidePage-box css-3oh5up-QxSidePage-box-QxModal-box"
      >
        <div
          class="QxHeaderBlock QxHeaderBlock_size_medium css-9oje8r-QxHeaderBlock-root"
        >
          <div
            class="QxHeaderBlock-content css-8aatdw-QxHeaderBlock-content"
          >
            <div
              class="QxHeaderBlock-title css-1p40syo-QxHeaderBlock-title"
            >
              Title
            </div>
            <div
              class="QxHeaderBlock-subTitle css-12h5hex-QxHeaderBlock-subTitle"
            >
              SubTitle
            </div>
          </div>
          <button
            class="QxBaseButton QxBaseButton_size_small QxBaseButton_color_secondary QxBaseButton_type_text QxBaseButton_borderRadius_medium QxIconButton QxIconButton_size_small QxHeaderBlock-closeButton css-yj9ie1-QxBaseButton-root-QxIconButton-root-QxHeaderBlock-closeButton"
            type="button"
          >
            <div
              class="QxIconButton-icon css-10kmd5c-QxIconButton-icon"
            >
              <svg
                fill="none"
                height="24"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M7.70714 17.7071C7.31661 18.0976 6.68345 18.0976 6.29292 17.7071C5.9024 17.3166 5.9024 16.6834 6.29292 16.2929L10.5858 12L6.29293 7.70711C5.9024 7.31658 5.9024 6.68342 6.29293 6.29289C6.68345 5.90237 7.31662 5.90237 7.70714 6.29289L12 10.5858L16.2929 6.29289C16.6834 5.90237 17.3166 5.90237 17.7071 6.29289C18.0977 6.68342 18.0977 7.31658 17.7071 7.70711L13.4142 12L17.7071 16.2929C18.0977 16.6834 18.0977 17.3166 17.7071 17.7071C17.3166 18.0976 16.6834 18.0976 16.2929 17.7071L12 13.4142L7.70714 17.7071Z"
                  fill="currentColor"
                />
              </svg>
            </div>
          </button>
        </div>
        <div
          class="QxModal-body css-r81a94-QxModal-body"
          style="min-height: NaNpx;"
        >
          Body
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`SidePage sidePage in the document 1`] = `
<body
  style=""
>
  <div>
    <div>
      <button
        class="QxBaseButton QxBaseButton_size_medium QxBaseButton_color_brand QxBaseButton_type_contained QxBaseButton_borderRadius_medium QxButton QxButton_size_medium css-3dyhpy-QxBaseButton-root-QxButton-root"
        data-testid="buttonTestId"
        type="button"
      />
    </div>
  </div>
</body>
`;

exports[`SidePage sidePage in the document 2`] = `
<body
  style="padding-right: 1024px; overflow: hidden;"
>
  <div>
    <div>
      <button
        class="QxBaseButton QxBaseButton_size_medium QxBaseButton_color_brand QxBaseButton_type_contained QxBaseButton_borderRadius_medium QxButton QxButton_size_medium css-3dyhpy-QxBaseButton-root-QxButton-root"
        data-testid="buttonTestId"
        type="button"
      />
    </div>
  </div>
  <div
    aria-hidden="false"
    class="QxOverScreen QxOverScreen_placement_right QxOverScreen_mounted QxModal QxModal_size_medium QxModal_scrollBehaviour_body QxModal_hasHeader QxModal_hasFooter QxSidePage css-1tn48qf-QxOverScreen-root-QxModal-root-QxSidePage-root"
    data-component="QxOverScreen"
    data-index="0"
    datatype="overscreen"
    role="presentation"
    tabindex="0"
  >
    <div
      aria-hidden="false"
      class="QxBackdrop css-lak5sv-QxBackdrop-root-QxBackdrop-root appear-active"
      data-component="QxBackdrop"
    />
    <div
      class="QxModal-scrollContainer QxSidePage-scrollContainer css-zezqa0-QxSidePage-scrollContainer-QxModal-scrollContainer-QxSidePage-scrollContainer-QxModal-scrollContainer appear-active"
      data-component="QxOverScreen-child"
    >
      <div
        class="QxModal-box QxSidePage-box css-3oh5up-QxSidePage-box-QxModal-box"
      >
        <div
          class="QxHeaderBlock QxHeaderBlock_size_medium css-9oje8r-QxHeaderBlock-root"
        >
          <div
            class="QxHeaderBlock-content css-8aatdw-QxHeaderBlock-content"
          >
            <div
              class="QxHeaderBlock-title css-1p40syo-QxHeaderBlock-title"
            >
              Title
            </div>
            <div
              class="QxHeaderBlock-subTitle css-12h5hex-QxHeaderBlock-subTitle"
            >
              SubTitle
            </div>
          </div>
          <button
            class="QxBaseButton QxBaseButton_size_small QxBaseButton_color_secondary QxBaseButton_type_text QxBaseButton_borderRadius_medium QxIconButton QxIconButton_size_small QxHeaderBlock-closeButton css-yj9ie1-QxBaseButton-root-QxIconButton-root-QxHeaderBlock-closeButton"
            type="button"
          >
            <div
              class="QxIconButton-icon css-10kmd5c-QxIconButton-icon"
            >
              <svg
                fill="none"
                height="24"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M7.70714 17.7071C7.31661 18.0976 6.68345 18.0976 6.29292 17.7071C5.9024 17.3166 5.9024 16.6834 6.29292 16.2929L10.5858 12L6.29293 7.70711C5.9024 7.31658 5.9024 6.68342 6.29293 6.29289C6.68345 5.90237 7.31662 5.90237 7.70714 6.29289L12 10.5858L16.2929 6.29289C16.6834 5.90237 17.3166 5.90237 17.7071 6.29289C18.0977 6.68342 18.0977 7.31658 17.7071 7.70711L13.4142 12L17.7071 16.2929C18.0977 16.6834 18.0977 17.3166 17.7071 17.7071C17.3166 18.0976 16.6834 18.0976 16.2929 17.7071L12 13.4142L7.70714 17.7071Z"
                  fill="currentColor"
                />
              </svg>
            </div>
          </button>
        </div>
        <div
          class="QxModal-body css-19ur24o-QxModal-body"
          style="min-height: NaNpx;"
        >
          Body
        </div>
        <div
          class="QxFooterBlock QxFooterBlock_size_medium QxFooterBlock_direction_vertical QxSidePage-footerBlock css-l1hdy0-QxSidePage-footerBlock-QxFooterBlock-root"
        >
          <button
            class="QxBaseButton QxBaseButton_size_medium QxBaseButton_color_brand QxBaseButton_type_contained QxBaseButton_borderRadius_medium QxButton QxButton_size_medium css-3dyhpy-QxBaseButton-root-QxButton-root"
            type="button"
          >
            <div
              class="QxButton-content css-ovifzf-QxButton-content"
            >
              Success
            </div>
          </button>
          <button
            class="QxBaseButton QxBaseButton_size_medium QxBaseButton_color_secondary QxBaseButton_type_outlined QxBaseButton_borderRadius_medium QxButton QxButton_size_medium css-aql2t3-QxBaseButton-root-QxButton-root"
            type="button"
          >
            <div
              class="QxButton-content css-ovifzf-QxButton-content"
            >
              Secondary
            </div>
          </button>
          <button
            class="QxBaseButton QxBaseButton_size_medium QxBaseButton_color_danger QxBaseButton_type_outlined QxBaseButton_borderRadius_medium QxButton QxButton_size_medium css-1fi0prz-QxBaseButton-root-QxButton-root"
            type="button"
          >
            <div
              class="QxButton-content css-ovifzf-QxButton-content"
            >
              Danger
            </div>
          </button>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`SidePage style params size medium 1`] = `
<body
  style="padding-right: 1024px; overflow: hidden;"
>
  <div />
  <div
    aria-hidden="false"
    class="QxOverScreen QxOverScreen_placement_right QxOverScreen_mounted QxModal QxModal_size_medium QxModal_scrollBehaviour_body QxModal_hasHeader QxSidePage css-1tn48qf-QxOverScreen-root-QxModal-root-QxSidePage-root"
    data-component="QxOverScreen"
    data-index="0"
    data-testid="testStyleParams-size-medium"
    datatype="overscreen"
    role="presentation"
    tabindex="0"
  >
    <div
      aria-hidden="false"
      class="QxBackdrop css-lak5sv-QxBackdrop-root-QxBackdrop-root appear-active"
      data-component="QxBackdrop"
    />
    <div
      class="QxModal-scrollContainer QxSidePage-scrollContainer css-zezqa0-QxSidePage-scrollContainer-QxModal-scrollContainer-QxSidePage-scrollContainer-QxModal-scrollContainer appear-active"
      data-component="QxOverScreen-child"
    >
      <div
        class="QxModal-box QxSidePage-box css-3oh5up-QxSidePage-box-QxModal-box"
      >
        <div
          class="QxHeaderBlock QxHeaderBlock_size_medium css-9oje8r-QxHeaderBlock-root"
        >
          <div
            class="QxHeaderBlock-content css-8aatdw-QxHeaderBlock-content"
          >
            <div
              class="QxHeaderBlock-title css-1p40syo-QxHeaderBlock-title"
            />
            <div
              class="QxHeaderBlock-subTitle css-12h5hex-QxHeaderBlock-subTitle"
            />
          </div>
          <button
            class="QxBaseButton QxBaseButton_size_small QxBaseButton_color_secondary QxBaseButton_type_text QxBaseButton_borderRadius_medium QxIconButton QxIconButton_size_small QxHeaderBlock-closeButton css-yj9ie1-QxBaseButton-root-QxIconButton-root-QxHeaderBlock-closeButton"
            type="button"
          >
            <div
              class="QxIconButton-icon css-10kmd5c-QxIconButton-icon"
            >
              <svg
                fill="none"
                height="24"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M7.70714 17.7071C7.31661 18.0976 6.68345 18.0976 6.29292 17.7071C5.9024 17.3166 5.9024 16.6834 6.29292 16.2929L10.5858 12L6.29293 7.70711C5.9024 7.31658 5.9024 6.68342 6.29293 6.29289C6.68345 5.90237 7.31662 5.90237 7.70714 6.29289L12 10.5858L16.2929 6.29289C16.6834 5.90237 17.3166 5.90237 17.7071 6.29289C18.0977 6.68342 18.0977 7.31658 17.7071 7.70711L13.4142 12L17.7071 16.2929C18.0977 16.6834 18.0977 17.3166 17.7071 17.7071C17.3166 18.0976 16.6834 18.0976 16.2929 17.7071L12 13.4142L7.70714 17.7071Z"
                  fill="currentColor"
                />
              </svg>
            </div>
          </button>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`SidePage style params size small 1`] = `
<body
  style="padding-right: 1024px; overflow: hidden;"
>
  <div />
  <div
    aria-hidden="false"
    class="QxOverScreen QxOverScreen_placement_right QxOverScreen_mounted QxModal QxModal_size_small QxModal_scrollBehaviour_body QxModal_hasHeader QxSidePage css-1tn48qf-QxOverScreen-root-QxModal-root-QxSidePage-root"
    data-component="QxOverScreen"
    data-index="0"
    data-testid="testStyleParams-size-small"
    datatype="overscreen"
    role="presentation"
    tabindex="0"
  >
    <div
      aria-hidden="false"
      class="QxBackdrop css-lak5sv-QxBackdrop-root-QxBackdrop-root appear-active"
      data-component="QxBackdrop"
    />
    <div
      class="QxModal-scrollContainer QxSidePage-scrollContainer css-zezqa0-QxSidePage-scrollContainer-QxModal-scrollContainer-QxSidePage-scrollContainer-QxModal-scrollContainer appear-active"
      data-component="QxOverScreen-child"
    >
      <div
        class="QxModal-box QxSidePage-box css-halxhf-QxSidePage-box-QxModal-box"
      >
        <div
          class="QxHeaderBlock QxHeaderBlock_size_small css-11mwf9r-QxHeaderBlock-root"
        >
          <div
            class="QxHeaderBlock-content css-8aatdw-QxHeaderBlock-content"
          >
            <div
              class="QxHeaderBlock-title css-1epjsp2-QxHeaderBlock-title"
            />
            <div
              class="QxHeaderBlock-subTitle css-q0oaiy-QxHeaderBlock-subTitle"
            />
          </div>
          <button
            class="QxBaseButton QxBaseButton_size_small QxBaseButton_color_secondary QxBaseButton_type_text QxBaseButton_borderRadius_medium QxIconButton QxIconButton_size_small QxHeaderBlock-closeButton css-yj9ie1-QxBaseButton-root-QxIconButton-root-QxHeaderBlock-closeButton"
            type="button"
          >
            <div
              class="QxIconButton-icon css-10kmd5c-QxIconButton-icon"
            >
              <svg
                fill="none"
                height="24"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M7.70714 17.7071C7.31661 18.0976 6.68345 18.0976 6.29292 17.7071C5.9024 17.3166 5.9024 16.6834 6.29292 16.2929L10.5858 12L6.29293 7.70711C5.9024 7.31658 5.9024 6.68342 6.29293 6.29289C6.68345 5.90237 7.31662 5.90237 7.70714 6.29289L12 10.5858L16.2929 6.29289C16.6834 5.90237 17.3166 5.90237 17.7071 6.29289C18.0977 6.68342 18.0977 7.31658 17.7071 7.70711L13.4142 12L17.7071 16.2929C18.0977 16.6834 18.0977 17.3166 17.7071 17.7071C17.3166 18.0976 16.6834 18.0976 16.2929 17.7071L12 13.4142L7.70714 17.7071Z"
                  fill="currentColor"
                />
              </svg>
            </div>
          </button>
        </div>
      </div>
    </div>
  </div>
</body>
`;
